File: JULY-2025/13-07-2025/p2410.swift
class Solution {
    func matchPlayersAndTrainers(_ players: [Int], _ trainers: [Int]) -> Int {
        let players = players.sorted()
        let trainers = trainers.sorted()
        let n = players.count
        let m = trainers.count
        var matches = 0
        var i = 0
        var j = 0
        
        while i < n && j < m {
            if players[i] <= trainers[j] {
                matches += 1
                i += 1
                j += 1
            }
            else { j += 1 }
        }
        
        return matches        
    }
}

File: JULY-2025/13-07-2025/p2410.rs
impl Solution {
    pub fn match_players_and_trainers(mut players: Vec<i32>, mut trainers: Vec<i32>) -> i32 {
        players.sort_unstable();
        trainers.sort_unstable();
        players
            .into_iter()
            .scan(trainers.into_iter(), |t, p| t.find(|&t| p <= t))
            .count() as _
    }
}